iina/FourCharCode+Extensions.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  FourCharCode+Extensions.swift
+//  iina
+//
+//  Created by Nate Weaver on 2020-03-08.
+//  Copyright © 2020 lhc. All rights reserved.
+//
+
+import Foundation
+
+extension FourCharCode: ExpressibleByStringLiteral {
+
+  public typealias StringLiteralType = String
+
+  public init(stringLiteral: String) {
+    // Match NSHFSTypeCodeFromFileType() behavior: Return 0 for invalid codes.
+    guard stringLiteral.count == 4, let data = stringLiteral.data(using: .macOSRoman) else {
```

Comment:
I think it might be a good idea to have initialization fail if these requirements aren't met. What do you think?

Reply:
I’m fine doing it that way; that was my initial thought, but then I decided to match the behavior of the notes function.

Reply:
I'd have to make it throw, since a failable initializer can't satisfy the protocol.